src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -432,16 +577,26 @@ def rmatmul(self, y):
         """
         raise NotImplementedError
 
-    def xor(self, y):
+    def xor(self, y: Union[int, torch.Tensor, "ReplicatedSharedTensor"]):
         """Apply the "xor" operation between "self" and "y".
 
         Args:
-            y: self^y
+            y: public bit
+
+        Returns:
+            ReplicatedSharedTensor: Result of the operation.
 
         Raises:
-            NotImplementedError: Raised when implementation not present
-        """
-        raise NotImplementedError
+            ValueError : If ring size is invalid.
+        """
+        if self.ring_size == 2:
+            return self + y
+        elif (
+            self.ring_size == PRIME_NUMBER or self.ring_size in RING_SIZE_TO_TYPE.keys()
```

Comment:
MAYBE: we could  do
`self.ring_size in RING_SIZE_TO_TYPE.keys()` only
as PRIME_NUMBER is also part of the keys.

Reply:
ok

Reply:
I think we can simply check:
`self.ring_size in RING_SIZE_TO_TYPE``` <-- this should automatically look only in the keys